src/amo/components/Link.js

```diff
@@ -1,5 +1,4 @@
-import path from 'path';
-
+import join from 'join-url';
```

Comment:
what about calling this `joinUrl`? Otherwise `join` seems a bit too generic as a global module var. Not a big deal, just something I noticed when reading the code.